### PR TITLE
config.tf: bump Kubernetes to 1.8.4

### PR DESCRIPTION
**Pull request by Quentin-M**
That be should all? However, there is no KVO image available yet for it.

/cc @sudhaponnaganti

### CODE REVIEW

**Comment by mxinden** [comment]
@Quentin-M once everything is in place, it might be helpful to add the `run-smoke-tests`, `run-conformance-tests` and `run-gui-tests` labels with all the available `platform/<xxx>` labels here. What do you think?

**Comment by Quentin-M** [comment]
@mxinden That's right, thank you! I'll do that once some of the remaining PRs and KVO gets in!

**Comment by diegs** [comment]
I was waiting on https://github.com/coreos/tectonic-installer/pull/2503

That needs to go into 1.8.4, and I'm not sure if that's the last console version.

cc @sudhaponnaganti @ggreer

**Comment by Quentin-M** [comment]
@diegs Ack. @ggreer was just saying they are building v2.6.1 right now for the next RC.

**Comment by yifan-gu** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

**Comment by Quentin-M** [comment]
ok to test

**Comment by yifan-gu** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

File: config.tf [hunk 1]
```diff
@@ -70,15 +70,15 @@ variable "tectonic_container_images" {
...
+    hyperkube                    = "quay.io/coreos/hyperkube:v1.8.4_coreos.0"
     identity                     = "quay.io/coreos/dex:v2.8.1"
     ingress_controller           = "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.9.0-beta.17"
     kenc                         = "quay.io/coreos/kenc:0.0.2"
     kubedns                      = "gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.5"
     kubednsmasq                  = "gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5"
     kubedns_sidecar              = "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5"
     kube_version                 = "quay.io/coreos/kube-version:0.1.0"
-    kube_version_operator        = "quay.io/coreos/kube-version-operator:v1.8.2-kvo.2"
+    kube_version_operator        = "quay.io/coreos/kube-version-operator:v1.8.4-kvo.1"
```
Comment (diegs) [inline config.tf]: Please bump to `v1.8.4-kvo.2`

**Comment by Quentin-M** [comment]
ok to test

**Comment by diegs** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

**Comment by sudhaponnaganti** [comment]
@diegs @yifan-gu - Can you check these failures. Need to get the build going.

**Comment by Quentin-M** [comment]
ok to. test

**Comment by yifan-gu** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

**Comment by yifan-gu** [comment]
retest this please

**Comment by yifan-gu** [comment]
ok to test

**Comment by yifan-gu** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

**Comment by yifan-gu** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

File: config.tf [hunk 2]
```diff
@@ -70,15 +70,15 @@ variable "tectonic_container_images" {
...
+    hyperkube                    = "quay.io/coreos/hyperkube:v1.8.4_coreos.0"
     identity                     = "quay.io/coreos/dex:v2.8.1"
     ingress_controller           = "quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.9.0-beta.17"
     kenc                         = "quay.io/coreos/kenc:0.0.2"
     kubedns                      = "gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.5"
     kubednsmasq                  = "gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5"
     kubedns_sidecar              = "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5"
     kube_version                 = "quay.io/coreos/kube-version:0.1.0"
-    kube_version_operator        = "quay.io/coreos/kube-version-operator:v1.8.2-kvo.2"
+    kube_version_operator        = "quay.io/coreos/kube-version-operator:v1.8.4-kvo.2"
```
Comment (yifan-gu) [inline config.tf]: v1.8.4-kvo.3 is out.

**Comment by sudhaponnaganti** [comment]
@Quentin-M @yifan-gu @cpanato - Is this ok to test now?

**Comment by sudhaponnaganti** [comment]
I guess my comment invoked the tests again

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
test upgrade from 1.7.9-tectonic.2 1.7.9-tectonic.2

**Comment by Quentin-M** [comment]
ok to test

**Comment by sudhaponnaganti** [comment]
@Quentin-M - Do you see the pr-merge is being successfully going? I see it taking too long

**Comment by Quentin-M** [comment]
@sudhaponnaganti Upstream conformance tests take 2hrs+ to run. For the other smoke tests, they should not be that long tho.

**Comment by cpanato** [comment]
@Quentin-M @sudhaponnaganti  smoke tests all passed, it is running the conformance now

**Comment by sudhaponnaganti** [comment]
@cpanato @Quentin-M  - Hmm I see smoke-tests failing

**Comment by sudhaponnaganti** [comment]
pr-merge also has a failure

**Comment by yifan-gu** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
AWS Canal:

```
Failures:

  1) aws-network-canal Interact with tectonic console can login in the tectonic console
     Failure/Error: expect(displayed?(USERNAME_INPUT)).to be_truthy

       expected: truthy value
            got: nil
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/aws/network_canal_spec.rb:6
     # ./lib/pages/login_page.rb:29:in `login_page'
     # ./lib/shared_examples/k8s.rb:104:in `block (3 levels) in <top (required)>'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:115:in `block in run'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `loop'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `run'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:33:in `block (2 levels) in setup'

  2) aws-network-canal Interact with tectonic console fail to login with wrong credentials
     Failure/Error: expect(displayed?(USERNAME_INPUT)).to be_truthy

       expected: truthy value
            got: nil
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/aws/network_canal_spec.rb:6
     # ./lib/pages/login_page.rb:29:in `login_page'
     # ./lib/shared_examples/k8s.rb:111:in `block (3 levels) in <top (required)>'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:115:in `block in run'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `loop'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `run'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /usr/local/bundle/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:33:in `block (2 levels) in setup'

Finished in 74 minutes 18 seconds (files took 0.62242 seconds to load)
9 examples, 2 failures
```

Azure basic:
```
Failures:

  1) azure-basic generates operator manifests
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  2) azure-basic verifies api checkpoint manifests
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  3) azure-basic verifies api secret checkpoints
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  4) azure-basic succeeds with the golang test suit
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  5) azure-basic installs the correct Container Linux version
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  6) azure-basic installs the correct Container Linux channel
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  7) azure-basic Interact with tectonic console can login in the tectonic console
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  8) azure-basic Interact with tectonic console fail to login with wrong credentials
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

  9) azure-basic scale up worker cluster can scale up nodes by 1 worker
     Failure/Error: raise "Aborting execution due startup failed. Error: #{e}"

     RuntimeError:
       Aborting execution due startup failed. Error: Error in wait_for_bootstrapping: timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     Shared Example Group: "withRunningClusterExistingBuildFolder" called from ./lib/shared_examples/k8s.rb:17
     Shared Example Group: "withRunningCluster" called from ./spec/azure/basic_spec.rb:6
     # ./lib/shared_examples/k8s.rb:27:in `rescue in block (2 levels) in <top (required)>'
     # ./lib/shared_examples/k8s.rb:23:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   timeout waiting for bootkube service to bootstrap on any of: ["52.161.105.110"]
     #   ./lib/cluster.rb:239:in `wait_for_service'

Finished in 51 minutes 28 seconds (files took 0.49133 seconds to load)
9 examples, 9 failures
```

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
From @sudhaponnaganti:

```
@quentin - can you merge the PR. the last check is successful but not exiting the job for some reason. 
If it is merged, I can make RC1 with known failures

Carlos will investigate the failures for smoke tests
```

**Comment by cpanato** [comment]
@Quentin-M @sudhaponnaganti the last run there are no smoke tests failure

**Pull request merged**